src/lib.rs

```diff
@@ -246,6 +247,32 @@ async fn install_dependency(
 
     janitor::cleanup_dependency(&dependency, false)?;
 
+    if config.recursive_deps {
+        if let Err(e) = install_subdependencies(&dependency) {
+            return Err(SoldeerError::DownloadError { dep: dependency.to_string(), source: e });
+        };
+    }
+
+    if !custom_url {
+        let new_dep = match get_url_type(dependency.url().unwrap()) {
+            UrlType::Git => Dependency::Git(GitDependency {
+                name: dependency.name().to_string(),
+                version: dependency.version().to_string(),
+                git: "".to_string(),
+                rev: Some(dependency.rev().unwrap().clone()),
+            }),
+            UrlType::Http => Dependency::Http(HttpDependency {
+                name: dependency.name().to_string(),
+                version: dependency.version().to_string(),
+                url: None,
+                checksum: None,
+            }),
+        };
+        add_to_config(&new_dep, &config_path).expect("add to config failed");
+    } else {
+        add_to_config(&dependency, &config_path).expect("add to config failed");
+    }
```

Comment:
This is a bit clunky. I would simply do the following (posting full function to avoid confusion):

```rust
async fn install_dependency(
    mut dependency: Dependency,
    regenerate_remappings: bool,
    recursive_deps: bool,
) -> Result<(), SoldeerError> {
    lock_check(&dependency, true)?;

    let config_path = match get_config_path() {
        Ok(file) => file,
        Err(e) => {
            cleanup_dependency(&dependency, true)?;
            return Err(e.into());
        }
    };

    let mut config = read_soldeer_config(Some(config_path.clone()))?;
    if regenerate_remappings {
        config.remappings_regenerate = regenerate_remappings;
    }

    if recursive_deps {
        config.recursive_deps = recursive_deps;
    }

    let result = download_dependency(&dependency, false)
        .await
        .map_err(|e| SoldeerError::DownloadError { dep: dependency.to_string(), source: e })?;
    match dependency {
        Dependency::Http(ref mut dep) => {
            dep.checksum = Some(result.hash);
            dep.url = Some(result.url);
        }
        Dependency::Git(ref mut dep) => dep.rev = Some(result.hash),
    }
    add_to_config(&dependency, &config_path)?;

    write_lock(&[dependency.clone()], LockWriteMode::Append)?;

    if let Dependency::Http(dep) = &dependency {
        if let Err(e) = unzip_dependency(dep) {
            cleanup_dependency(&dependency, true)?;
            return Err(SoldeerError::DownloadError { dep: dependency.to_string(), source: e });
        }
    }

    janitor::healthcheck_dependency(&dependency)?;

    janitor::cleanup_dependency(&dependency, false)?;

    if config.recursive_deps {
        if let Err(e) = install_subdependencies(&dependency) {
            return Err(SoldeerError::DownloadError { dep: dependency.to_string(), source: e });
        };
    }

    if config.remappings_generate {
        if config_path.to_string_lossy().contains("foundry.toml") {
            match config.remappings_location {
                RemappingsLocation::Txt => {
                    remappings_txt(&RemappingsAction::Add(dependency), &config_path, &config)
                        .await?
                }
                RemappingsLocation::Config => {
                    remappings_foundry(&RemappingsAction::Add(dependency), &config_path, &config)
                        .await?
                }
            }
        } else {
            remappings_txt(&RemappingsAction::Add(dependency), &config_path, &config).await?;
        }
    }

    Ok(())
}
```